### DISCUSSION TITLE
posts/2023-06-03-searching-for-user-sessions/

### DISCUSSION THREAD

**Discussion by giscus** [L-Groeninger/lukas-r.blog#35] [Comments]
# posts/2023-06-03-searching-for-user-sessions/

How to assign Session ID's to group event data.

https://lukas-r.blog/posts/2023-06-03-searching-for-user-sessions/

**Comment by einarhjorleifsson** [comment]
thnx for this. seems like adding a group_by (here on user_id, but multiple groups are also kosher) achieves the same results:
searches |> 
  group_by(user_id) |> 
  mutate(time_dif = c(0, diff(created_at)),
         time_dif_reset = if_else(time_dif >= 10, TRUE, FALSE),
         reset = if_else(time_dif_reset, TRUE, FALSE),
         session_id = cumsum(reset))

**Reply by L-Groeninger** [reply]
Thanks:) 
This would however only assign a unique session_id per user right? The session_id column would not be unique

**Reply by einarhjorleifsson** [reply]
oh, yes - should have mentioned that - unique within the grouping, whatever that may be.
